crates/stackable-operator/src/role_utils.rs

```diff
@@ -144,10 +155,129 @@ pub struct CommonConfiguration<T> {
+    /// for details on the usage.
+    #[serde(default)]
+    pub jvm_argument_overrides: JvmArgumentOverrides,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JvmArgumentOverrides {
+    /// JVM arguments to be added
+    #[serde(default)]
+    add: Vec<String>,
+
+    /// JVM arguments to be removed by exact match
+    //
+    // HashSet to be optimized for quick lookup
+    #[serde(default)]
+    remove: HashSet<String>,
+
+    /// JVM arguments matching any of this regexes will be removed
+    #[serde(default)]
+    remove_regex: Vec<String>,
+}
+
+impl JvmArgumentOverrides {
+    pub fn new(add: Vec<String>, remove: HashSet<String>, remove_regex: Vec<String>) -> Self {
+        Self {
+            add,
+            remove,
+            remove_regex,
+        }
+    }
+
+    pub fn new_with_only_additions(add: Vec<String>) -> Self {
+        Self {
+            add,
+            ..Default::default()
+        }
+    }
+
+    /// Called on **merged** [`JvmArgumentOverrides`}, returns all arguments that should be passed to the JVM.
+    ///
+    /// **Can only be called on merged config, it will panic otherwise!**
+    ///
+    ///  We are panicking (instead of returning an Error), because this is not the users fault, but
+    /// the operator is doing things wrong
+    pub fn effective_jvm_config_after_merging(&self) -> &Vec<String> {
+        assert!(
+            self.remove.is_empty(),
+            "After merging there should be no removals left. \"effective_jvm_config_after_merging\" should only be called on merged configs!"
+        );
+        assert!(
+            self.remove_regex.is_empty(),
+            "After merging there should be no removal regexes left. \"effective_jvm_config_after_merging\" should only be called on merged configs!"
+        );
+
+        &self.add
+    }
+}
+
+impl Merge for JvmArgumentOverrides {
+    /// Please watch out: Merge order is complicated for this merge.
+    /// Test your code!
+    #[instrument]
+    fn merge(&mut self, defaults: &Self) {
+        let regexes = self
+            .remove_regex
+            .iter()
+            .filter_map(|regex| {
+                let without_anchors = regex.trim_start_matches('^').trim_end_matches('$');
+                let with_anchors = format!("^{without_anchors}$");
+                match Regex::new(&with_anchors) {
+                    Ok(regex) => Some(regex),
+                    Err(error) => {
+                        tracing::warn!(
+                            regex,
+                            regex_without_anchors = without_anchors,
+                            regex_with_anchors = with_anchors,
+                            ?error,
+                            "Could not parse regex from \"jvmArgumentOverrides.removeRegex\", ignoring it"
```

Comment:
Im not sure its a good idea to ignore the regex instead of erroring?

Is this conscious decision or due to the trait?

Reply:
This is mostly because the trait is not fallible. We could not use the Trait, but not sure which is better ^^
Any opinions?

We can also bump warn to error maybe

Reply:
Hmm i think i prefer to error out here (with a Result) instead of deploying a cluster that may or may not do what its supposed to?

Reply:
Updated the PR to not use the Merge trait any more, but a fallible function